anaUltraScurve.py

```diff
@@ -210,16 +226,61 @@ def overlay_fit(VFAT, CHAN):
     trimrange_list[event.vfatN][event.vfatCH] = event.trimRange
     pass
```

Comment:
Running over all events only once is possible but would be a huge change by itself. I suggest to leave this optimization to another PR in order to make reviewing changes easier.

Getting the `strip` and the `pan_pin` for the event can only be done on an event-by-event basis. I don't see any other way to do it except always filling the histogram per-channel and shuffling bins afterwards.

Lines 224-226 can be put in any of the two "event loops", I don't see any good reason to touch these lines.